cupy/lib/_routines_poly.py

```diff
@@ -104,6 +105,124 @@ def polymul(a1, a2):
+        deg (int): degree of the fitting polynomial.
+        rcond (float, optional): relative condition number of the fit.
+            The default value is len(x) * eps.
+        full (bool, optional): indicator of the return value nature.
+            When False (default), only the coefficients are returned.
+            When True, diagnostic information is also returned.
+        w (cupy.ndarray, optional): weights applied to the y-coordinates
+            of the sample points of shape (M, ).
+        cov (bool or str, optional): if given, returns the coefficients
+            along with the covariance matrix.
+
+    Returns:
+        cupy.ndarray: of shape (deg + 1,) or (deg + 1, K).
+            Polynomial coefficients from highest to lowest degree
+        tuple (cupy.ndarray, int, cupy.ndarray, float):
+            Present only if ``full`` = True.
+            Sum of squared residuals of the least-squares fit,
+            rank of the scaled Vandermonde coefficient matrix,
+            its singular values, and the specified value of ``rcond``.
+        cupy.ndarray: of shape (M, M) or (M, M, K).
+            Present only if ``full`` = False and ``cov`` = True.
+            The covariance matrix of the polynomial coefficient estimates.
+
+    .. warning::
+
+        numpy.RankWarning: The rank of the coefficient matrix in the
+        least-squares fit is deficient. It is raised if ``full`` = False.
+
+    .. seealso:: :func:`numpy.polyfit`
+
+    """
+    if isinstance(x, cupy.poly1d):
+        x = x.coeffs
+    if isinstance(y, cupy.poly1d):
+        y = y.coeffs
+
+    x = _astype(x)
+    y = _astype(y)
+    deg = int(deg)
+
+    if deg < 0:
+        raise ValueError('expected deg >= 0')
+    if x.ndim != 1:
+        raise TypeError('expected 1D vector for x')
+    if x.size == 0:
+        raise TypeError('expected non-empty vector for x')
+    if y.ndim < 1 or y.ndim > 2:
+        raise TypeError('expected 1D or 2D array for y')
+    if x.size != y.shape[0]:
+        raise TypeError('expected x and y to have same length')
+
+    lhs = cupy.polynomial.polynomial.polyvander(x, deg)[:, ::-1]
+    rhs = y
+
+    if w is not None:
+        w = _astype(w)
+        if w.ndim != 1:
+            raise TypeError('expected a 1-d array for weights')
+        if w.size != x.size:
+            raise TypeError('expected w and y to have the same length')
+
+        lhs *= w[:, None]
+        if rhs.ndim == 2:
+            w = w[:, None]
+        rhs *= w
+
+    if rcond is None:
+        rcond = x.size * cupy.finfo(x.dtype).eps
+
+    scale = cupy.sqrt((cupy.square(lhs)).sum(axis=0))
+    lhs /= scale
+    c, resids, rank, s = cupy.linalg.lstsq(lhs, rhs, rcond)
+    c = (c.T / scale).T
+
+    order = deg + 1
+    if rank != order and not full:
+        msg = 'Polyfit may be poorly conditioned'
+        warnings.warn(msg, numpy.RankWarning, stacklevel=4)
+
+    if y.ndim != 1:
+        c = cupy.ascontiguousarray(c)
```

Comment:
Are these lines for the compatibility of the strides of return values? If so, please check it in the tests. `testing.assert_allclose` does not check the equalities of the strides of the inputs in default.

Reply:
You can pass `contiguous_check` option to `testing.numpy_cupy_allclose`.